pygdf/numerical.py

```diff
@@ -191,3 +197,9 @@ def numeric_column_unaryop(operand, op, out_dtype):
 def numeric_column_compare(lhs, rhs, op):
     return numeric_column_binop(lhs, rhs, op, out_dtype=np.bool_)
 
+
+def numeric_normalize_types(*args):
+    """Cast all args to a common type using numpy promotion logic
+    """
+    dtype = np.result_type(*[a.dtype for a in args])
```

Comment:
Thankfully, pandas doesn't use unsigned dtypes, so we are unlikely to see the really bizarre type promotion cases here.